test/Core.Test/Services/CollectionServiceTests.cs

```diff
@@ -1,184 +1,166 @@
+            Assert.True(collection.CreationDate - utcNow < TimeSpan.FromSeconds(1));
+            Assert.True(collection.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
         }
 
-        [Fact]
-        public async Task SaveAsync_CollectionNotFound()
+        [Theory, CollectionDefaultIdAutoData]
+        public async Task SaveAsync_DefaultIdWithGroups_CreateCollectionWithGroupsInRepository(Collection collection,
+            IEnumerable<SelectionReadOnly> groups, Organization organization, OrganizationUser organizationUser,
+            SutProvider<CollectionService> sutProvider)
         {
-            var collectionService = new CollectionService(
-                _eventService,
-                _organizationRepository,
-                _organizationUserRepository,
-                _collectionRepository,
-                _userRepository,
-                _mailService);
+            organization.UseGroups = true;
+            sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(organization.Id).Returns(organization);
+            var utcNow = DateTime.UtcNow;
 
-            var id = Guid.NewGuid();
+            await sutProvider.Sut.SaveAsync(collection, groups);
 
-            var collection = new Core.Models.Table.Collection
-            {
-                Id = id,
-            };
+            await sutProvider.GetDependency<ICollectionRepository>().Received().CreateAsync(collection, groups);
+            await sutProvider.GetDependency<IEventService>().Received()
+                .LogCollectionEventAsync(collection, EventType.Collection_Created);
+            Assert.True(collection.CreationDate - utcNow < TimeSpan.FromSeconds(1));
+            Assert.True(collection.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
+        }
 
-            var ex = await Assert.ThrowsAsync<BadRequestException>(() => collectionService.SaveAsync(collection));
+        [Theory, CollectionNonDefaultIdAutoData]
+        public async Task SaveAsync_NonDefaultId_ReplacesCollectionInRepository(Collection collection, Organization organization, SutProvider<CollectionService> sutProvider)
+        {
+            var creationDate = collection.CreationDate;
+            sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(organization.Id).Returns(organization);
+            var utcNow = DateTime.UtcNow;
+
+            await sutProvider.Sut.SaveAsync(collection);
+
+            await sutProvider.GetDependency<ICollectionRepository>().Received().ReplaceAsync(collection);
+            await sutProvider.GetDependency<IEventService>().Received()
+                .LogCollectionEventAsync(collection, EventType.Collection_Updated);
+            Assert.True(collection.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
+            Assert.Equal(collection.CreationDate, creationDate);
+            Assert.True(collection.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
```

Comment:
Was this assertion intended to be different here or duplicated? Perhaps asserting the revision date is no longer = to the creation date?

Reply:
I accidentally duplicated it. There's already an assertion for `collection.CreationDate` to not have been changed so I think removing this line should be sufficient.